<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TouchPageWidget</class>
 <widget class="QWidget" name="TouchPageWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>663</width>
    <height>418</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_9">
   <item>
    <widget class="KMessageWidget" name="gesturesWarning">
     <property name="text">
      <string>Enabling gestures may break touch input using a touchscreen.</string>
     </property>
     <property name="visible">
      <bool>false</bool>
     </property>
     <property name="closeButtonVisible">
      <bool>false</bool>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="messageType">
      <enum>KMessageWidget::Warning</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="touchCheckBox">
     <property name="toolTip">
      <string>If enabled the tablet will emit touch events.</string>
     </property>
     <property name="text">
      <string>Enable Touch</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="touchHLayout1">
     <item>
      <widget class="QGroupBox" name="trackingModeGroupBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Defines how the cursor movement mode will be used.</string>
       </property>
       <property name="title">
        <string>Touch Tracking Mode</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QRadioButton" name="trackAbsoluteRadioButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>&amp;Map to Screen (Absolute Mode)</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="trackRelativeRadioButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Map &amp;to Cursor (Relative Mode)</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="trackingFreeSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::MinimumExpanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>1</width>
            <height>5</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="touchMappingGroupBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Touch Mapping</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QPushButton" name="touchMappingTabletButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Allows one to specify the tablet area for different screen selections in absolute cursor mode.</string>
          </property>
          <property name="text">
           <string>Map Touch Area to Screen</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="trackingWarningLayout">
          <item>
           <widget class="QLabel" name="trackingWarningIcon">
            <property name="minimumSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="trackingWarningLabel">
            <property name="text">
             <string>Your currently selected mapping can not be applied in relative tracking mode.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="touchMappingFreeSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::MinimumExpanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>1</width>
            <height>5</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QCheckBox" name="gesturesCheckBox">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>If enabled the touch events can recognize gestures.</string>
     </property>
     <property name="text">
      <string>Enable Gestures</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="touchHLayout2">
     <item>
      <widget class="QGroupBox" name="gesturesGroupBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Gestures</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="2" column="0">
         <widget class="QLabel" name="scrollDistanceLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Minimum motion before sending a scroll gesture.</string>
          </property>
          <property name="text">
           <string>Scroll Distance</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="scrollDistanceSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>The minimum motion in tablet units before sending a scroll event.</string>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="zoomDistanceLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Minimum distance for a zoom gesture is recognized.</string>
          </property>
          <property name="text">
           <string>Zoom Distance</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="zoomDistanceSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>The minimum distance in tablet units for a zoom gesture to be recognized.</string>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="tapTimeLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Minimum time between taps for a right click.</string>
          </property>
          <property name="text">
           <string>Tap Time</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="tapTimeSpinBox">
          <property name="toolTip">
           <string>The minimum time in milliseconds between taps to trigger a right click.</string>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="scrollInversionLabel">
          <property name="text">
           <string>Invert Scroll Direction</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="scrollInversionCheckBox">
          <property name="toolTip">
           <string>If selected, the scroll direction for the scroll gesture is inverted.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="dummyWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalFreeSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>1</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header>kmessagewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>touchCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onTouchModeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trackAbsoluteRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onTrackingModeAbsolute(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trackRelativeRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onTrackingModeRelative(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>touchMappingTabletButton</sender>
   <signal>clicked()</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onTabletMappingClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>302</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scrollDistanceSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onProfileChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>375</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomDistanceSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onProfileChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>401</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tapTimeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onProfileChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gesturesCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onGesturesModeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scrollInversionCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>TouchPageWidget</receiver>
   <slot>onProfileChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onTouchModeChanged(int)</slot>
  <slot>onGesturesModeChanged(int)</slot>
  <slot>onProfileChanged()</slot>
  <slot>onTabletMappingClicked()</slot>
  <slot>onTrackingModeAbsolute(bool)</slot>
  <slot>onTrackingModeRelative(bool)</slot>
 </slots>
</ui>
